src/Package/Impl/ProjectSystem/ProjectProperties.cs

```diff
@@ -96,6 +104,111 @@ public ProjectProperties(ConfiguredProject configuredProject)
+        /// </summary>
+        /// <remarks>
+        /// Default destination is ~/RTVSProjects/.
+        /// </remarks>
+        public async Task SetRemoteProjectPathAsync(string remoteProjectPath) {
+            var runProps = await this.GetConfigurationRunPropertiesAsync();
+            await runProps.RemoteProjectPath.SetValueAsync(remoteProjectPath);
+        }
+
+        /// <summary>
+        /// Gets the filter string that selects files to be sent to remote host.
+        /// </summary>
+        /// <remarks>
+        /// Default filter string is ".r;.rmd;.rsettings". This selects all R script files, R markdown, 
+        /// and R Settings files.
+        /// </remarks>
+        public async Task<string> GetFileFilterAsync() {
+            var runProps = await this.GetConfigurationRunPropertiesAsync();
+            var filter = await runProps.TransferFilesFilter.GetEvaluatedValueAsync();
+            if (string.IsNullOrWhiteSpace(filter)) {
+                return ".r;.rmd;.rsettings";
+            }
+            return filter;
+        }
+
+        /// <summary>
+        /// Gets the filter string that selects files to be sent to remote host.
+        /// </summary>
+        /// <remarks>
+        /// Default filter string is ".r;.rmd;.rsettings". This selects all R script files, R markdown, 
+        /// and R Settings files.
+        /// </remarks>
+        public async Task SetFileFilterAsync(string fileTransferFilter) {
+            var runProps = await this.GetConfigurationRunPropertiesAsync();
+            await runProps.TransferFilesFilter.SetValueAsync(fileTransferFilter);
+        }
+
+        /// <summary>
+        /// Defines if project has to be transfered to remote host during run.
+        /// </summary>
+        public async Task<bool> GetTransferProjectOnRunAsync() {
+            var runProps = await this.GetConfigurationRunPropertiesAsync();
+            var val = await runProps.TransferProjectOnRun.GetEvaluatedValueAsync();
+            return ParseBooleanProperty(val, false);
+        }
+
+        /// <summary>
+        /// Defines if project has to be transfered to remote host during run.
+        /// </summary>
+        public async Task SetTransferProjectAsync(bool val) {
+            var runProps = await this.GetConfigurationRunPropertiesAsync();
+            await runProps.TransferProjectOnRun.SetValueAsync(val);
+        }
+
+        /// <summary>
+        /// Gets all the R script files in the current project.
+        /// </summary>
+        public IEnumerable<string> GetRFilePaths() {
+            IList<string> files = new List<string>();
+            string projectDir = Path.GetDirectoryName(this.ConfiguredProject.UnconfiguredProject.FullPath);
+            var projDir = _fileSystem.GetDirectoryInfo(projectDir);
+
+            Stack<IDirectoryInfo> dirs = new Stack<IDirectoryInfo>();
+            dirs.Push(projDir);
+
+            while (dirs.Count > 0) {
```

Comment:
It might be worth making this (enumerating files recursively) a helper extension method on `IDirectoryInfo`.